src/pickup.cpp

```diff
@@ -251,6 +251,116 @@ static bool select_autopickup_items( std::vector<item> &here, std::vector<bool>
+            return true;
+        }
+    }
+
+    if ( !p.can_pickup(false) ) {
+        if ( p.is_player() && !silent_chk ) {
+            add_msg( m_info, _(("You cannot pick up items with your %s! " + append_fail_msg).c_str()), p.weapon.tname().c_str() );
+        }
+        return false;
+    }
+
+    if ( newit.made_of(LIQUID) ) {
+        if ( p.is_player() && !silent_chk ) {
+            add_msg( m_info, _(("You can't pick up a liquid! " + append_fail_msg).c_str()) );
+        }
+        return false;
+    }
+
+    if ( !p.can_pickWeight(newit.weight(), false) ) {
+        if ( p.is_player() && !silent_chk ) {
+            add_msg( m_info, _(("The %s is too heavy! " + append_fail_msg).c_str()), newit.display_name().c_str() );
+        }
+        return false;
+    }
+
+    // check if there is a quiver that can store at least 1 arrow/bolt
+    if ( newit.is_ammo() && (newit.ammo_type() == "arrow" || newit.ammo_type() == "bolt") ) {
+        std::vector<item *>quivers;
+        for ( auto & worn_item : p.worn ) {
+            if( worn_item.type->can_use( "QUIVER") ) {
+                quivers.push_back( &worn_item );
+            }
+        }
+
+
+        bool have_space_in_quiver = false;
+
+        for ( std::vector<item *>::iterator it = quivers.begin(); it != quivers.end(); it++ ) {
+            item *quiver = *it;
+
+            long max_arrows = (long) quiver->max_charges_from_flag( "QUIVER" );
+
+            if ( quiver->contents.empty() || (quiver->contents[0].charges < max_arrows && quiver->contents[0].type->id == newit.type->id) ) {
+                have_space_in_quiver = true;
+                break;
+            }
+        }
+
+        if ( !have_space_in_quiver && !p.can_pickVolume(newit.volume()) ) {
+                if ( p.is_player() && !silent_chk ) {
+                    add_msg( m_info, _(("There's no room in your inventory for the %s. " + append_fail_msg).c_str()),
+                                        newit.tname(newit.charges).c_str() );
+                }
+                return false;
+        }
+
+        if ( have_space_in_quiver ) {
+            return true;
+        }
+    } // end if( newit.is_ammo() && (newit.ammo_type() == "arrow" || newit.ammo_type() == "bolt"))
+
+    if ( !p.can_pickVolume(newit.volume()) ) {
+        if ( chk_armor && newit.is_armor() ) {
+            return true;
+        }
+
+        if ( allow_swap ) {
+            if ( chk_keep_hands_free && p.keep_hands_free && !p.is_armed() ) {
+                if ( p.is_player() && !silent_chk ) {
+                    add_msg( m_info, _(("There's no room in your inventory for the %s "
+                                       "and you have decided to keep your hands free. " + append_fail_msg).c_str()),
+                                        newit.display_name().c_str() );
+                }
+                return false;
+            }
+
+            if ( !p.is_armed() || (p.is_armed() && !p.weapon.has_flag("NO_UNWIELD")) ) {
```

Comment:
This `if` seems to have no meaning. If the condition is `true`, the function returns `true` (next line), if the condition is `false`, it falls through to the end of the function and returns `true` from there.